File: API/Helpers/UserParams.cs
namespace API.Helpers;

public class UserParams : PaginationParams
{
  
    public string CurrentUsername { get; set; }
    public string Genre { get; set; }
    public int MinAge { get; set; } = 1;
    public int MaxAge { get; set; } = 15;
    public string OrderBy { get; set; } = "lastActive";

}

File: client/src/app/_models/userParams.ts
import { User } from "./user";

export class UserParams {
    genre: string;
    minAge = 1;
    maxAge = 15;
    pageNumber = 1;
    pageSize = 8;
    orderBy = 'lastActive';

    constructor(user: User){
        this.genre = user.genre === "Female" ? "Male" : "Female";
    }

}